engine/src/vault/crypto_box.rs

```diff
@@ -36,11 +42,11 @@ pub trait BoxProvider: 'static + Sized + Ord + PartialOrd {
 }
 
 /// A key to the crypto box.  [`Key`] is stored on the heap which makes it easier to erase. Makes use of the
-/// [`GuardedVec<u8>`] type to protect the data.
+/// [`Buffer<u8>`] type to protect the data.
 #[derive(Serialize, Deserialize)]
 pub struct Key<T: BoxProvider> {
     /// the guarded raw bytes that make up the key
-    pub key: GuardedVec<u8>,
+    pub key: Buffer<u8>,
```

Comment:
I know in practice it does not make a difference, but should we use the `RamMemory` type here because the `Key` is stored for a longer period of time? I thought the `Buffer` type should only be used when accessing a value in a `LockedMemory`.